src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java

```diff
@@ -171,11 +171,12 @@ public void setHeader(String header) {
 
     /**
      * Load header to check against from a Reader into readerLines.
+     * This method has package access modifier because it is used in tests.
      *
      * @param headerReader delivers the header to check against
      * @throws IOException if
      */
-    private void loadHeader(final Reader headerReader) throws IOException {
+    /* package */ void loadHeader(final Reader headerReader) throws IOException {
```

Comment:
This is the only possible way. See https://github.com/mockito/mockito/wiki/Mockito-And-Private-Methods

Reply:
This worries me a lot ... We did a lot of work to remove package and protected visibility just testing.

Reply:
I have an idea. Maybe it is possible to implement a mock chain to simulate an exception in the `loadHeader` method.

Reply:
Succeeded with
```java
            final Exception ex = assertThrows(Exception.class,
                    () -> {
                        try (MockedConstruction<LineNumberReader> mocked = mockConstruction(
                                LineNumberReader.class, (mock, context) -> {
                                    when(mock.readLine()).thenThrow(IOException.class);
                                })) {
                            check.setHeader("header");
                        }
                    },
                    "IllegalArgumentException expected");
```
This PR will be closed, the changes will be in the PR #10947